packages/dx-grid-core/src/plugins/sorting-state/reducers.ts

```diff
@@ -9,10 +9,10 @@ export const changeColumnSorting: PureReducer<ColumnSortingState, ChangeSortingP
 
   let nextSorting: any[] = [];
   if (keepOther === true) {
```

Comment:
If `keepOther` is `true` and neither `columnSortingIndex > -1` nor `direction !== null` are `true` then function result is result is falsely different.